manager/src/main/java/io/gingersnapproject/Maps.java

```diff
@@ -0,0 +1,45 @@
+package io.gingersnapproject;
+
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.stream.Stream;
+
+import javax.inject.Singleton;
+
+import org.infinispan.commons.marshall.WrappedByteArray;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+
+@Singleton
+public class Maps {
+   private final ConcurrentMap<String, com.github.benmanes.caffeine.cache.Cache<WrappedByteArray, WrappedByteArray>> maps = new ConcurrentHashMap<>();
+   private final ConcurrentMap<String, Cache<WrappedByteArray, List<byte[]>>> multiMaps = new ConcurrentHashMap<>();
```

Comment:
We don't have to, I was debating making it part of the same map. I figured it would be simpler to do it this way though so it is more clearly defined what is a Map and what is a MultiMap. But I can be convinced to change it, I personally don't feel it will save that much code/maintenance myself and if fact will be a bit uglier as we have to use an instanceof/cast for the value.

Reply:
And the problem is the way the schema works in debezium is they are effectively appending the schema updates. We need a nice way to only write the single schema update but fetch all of them, it seemed like a natural fit for multi map.

Reply:
No objections from me if it's the simplest solution. There's no point removing this HR code to simply the cache-manager if it's going to make the db-syncer integration more complex.